### DISCUSSION TITLE
misc ideas

### DISCUSSION THREAD

**Discussion by SomeGuyNamedMay** [nix-community/stylix#24] [Ideas]
Ive got a few ideas as to extra features that id like to see come to stylix. Im willing to implement them myself, obviously, but id like to make sure that your actually interested in adding these to the project.

1. polarity based on time of day.
2. managing of icon themes through stylix, ie automaticly configuring rofi or other such launchers to use the correct icon theme.
3. setting the wallpaper as an animated video via mpvpaper as an option for wayland.
4. hyprland support.
5. waybar support.

**Comment by danth** [comment]
1 isn't really possible, as most apps don't update their theme until they are restarted.

There is already an option in Home Manager related to 2, however we could look into adding it to Stylix even if its just an alias for the time being.

3 is totally possible, perhaps an option like `stylix.animation` which is used as the background where possible, falling back to `stylix.image` if it's undefined. Then `stylix.image` would default to the first frame of the animation, to be used where animations aren't supported.

4 and 5 are just adding extra modules which should be fairly simple.

You might want to wait for #23 before making any major contributions, as it rearranges a lot of code.

**Reply by jakubkaczor** [reply]
I believe no one expects changing the color scheme on the fly for every application. I would be very contented with just the config update, but some applications allow for dynamic config reload for all instances with a command, so this could be a nice addition in particular cases.
To achieve a similar functionality, I currently configure software in Nix to load configuration files outside of Nix store that I dynamically change using [darkman](https://darkman.whynothugo.nl/). More general and appropriate approach would be to generate a systemd service with a timer that switches symlinks for configuration files between two versions, one for the light theme and one for the dark theme. I consider it more general, because it would implement the feature even for programs that cannot include other files in their configuration files.
For me, what is described in 1. would be the killer feature.
